geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -119,9 +124,12 @@ public void fromData(DataInput in, DeserializationContext context)
     super.fromData(in, context);
     int size = InternalDataSerializer.readPrimitiveInt(in);
     members = new Object2ObjectOpenCustomHashMap<>(size, ByteArrays.HASH_STRATEGY);
+    scoreSet = new OrderStatisticsTree<>();
     for (int i = 0; i < size; i++) {
-      members.put(InternalDataSerializer.readByteArray(in),
-          InternalDataSerializer.readByteArray(in));
+      byte[] member = InternalDataSerializer.readByteArray(in);
+      byte[] score = InternalDataSerializer.readByteArray(in);
+      members.put(member, score);
+      scoreSet.add(new OrderedSetEntry(member, score));
```

Comment:
is there a chance that adding the new OrderedSetEntry to the OrderStatisticsTree could fail? If it does, then the value needs to be cleaned up from the map before throwing up the exception

Reply:
Yes, depending on how we filter out invalid floats. We could add it to the members map, then fault out adding it to the scoreSet. Swapping the order of which one we add _should_ avoid that.

Reply:
I added some test conditions to validate that if adding fails due to invalid doubles, we confirm it's not still lingering in the map to cause trouble.